### PR TITLE
Use gradle/actions/setup-gradle instead of custom cache

### PR DESCRIPTION
**Pull request by BraisGabin**

### CODE REVIEW

File: .github/workflows/publish-snapshot.yml
Comment (TWiStErRob) [inline .github/workflows/publish-snapshot.yml:18]: The tests **before merge** should run on the same JDK that publishes, to prevent surprises when releasing, which is **after merge**. (i.e. add setup-java to the other workflow)

**Review by TWiStErRob — APPROVED** [review]

File: .github/workflows/merge-check.yml
Comment (TWiStErRob) [inline .github/workflows/merge-check.yml:44]: Hmm, wait, I think we need still this, because this is not a Gradle controlled folder! 

https://github.com/gradle/gradle-build-action?tab=readme-ov-file#which-content-is-cached

File: .github/workflows/publish-snapshot.yml
Comment (BraisGabin) [inline .github/workflows/publish-snapshot.yml:18]:
```suggestion
          distribution: 'zulu'
```

File: .github/workflows/publish-snapshot.yml
Comment (BraisGabin) [reply .github/workflows/publish-snapshot.yml]: I did that change just to align with the distribution that we use in the main repo. But this change it not related with this PR so I just reverted it.

File: .github/workflows/merge-check.yml
Comment (BraisGabin) [reply .github/workflows/merge-check.yml]: I'm going to wait for this decision first: https://github.com/detekt/sarif4k/pull/86#issuecomment-1921842131 then I'll go back here.

File: .github/workflows/publish-snapshot.yml
Comment (TWiStErRob) [reply .github/workflows/publish-snapshot.yml]: Strong no! Temurin is pre-installed:
https://github.com/TWiStErRob/github-actions-test/actions/runs/7746795967/job/21125817607#step:3:1

Comment (TWiStErRob) [reply .github/workflows/publish-snapshot.yml]: ah, ok, re  https://github.com/detekt/sarif4k/pull/81#discussion_r1474847372, will raise another PR

Comment (TWiStErRob) [reply .github/workflows/publish-snapshot.yml]: https://github.com/detekt/sarif4k/pull/88

**Comment by BraisGabin** [comment]
Ok, now this should be ready to merge.

**Review by TWiStErRob — APPROVED** [review]

**Pull request merged**
